Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -0,0 +1,130 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "DetectorsBase/GeometryManager.h"
+#include "TRDBase/TrackletTransformer.h"
+#include "TMath.h"
+
+using namespace o2::trd;
+
+TrackletTransformer::TrackletTransformer()
+{
+  o2::base::GeometryManager::loadGeometry();
+  mGeo = o2::trd::Geometry::instance();
+  mGeo->createPadPlaneArray();
+  mGeo->createClusterMatrixArray();
+}
+
+void TrackletTransformer::loadPadPlane(int hcid)
+{
+  int detector = hcid / 2;
+  int stack = mGeo->getStack(detector);
+  int layer = mGeo->getLayer(detector);
+  mPadPlane = mGeo->getPadPlane(layer, stack);
+}
+
+float TrackletTransformer::calculateX()
+{
+  // report x at 5mm below the cathode plan to minimize error propogration from tracklet fit and driftV calibration
+  float driftHeight = mGeo->cdrHght();
+  float x = driftHeight - 0.5;
+
+  return x;
+}
+
+float TrackletTransformer::calculateY(int hcid, int column, int position)
+{
+  double padWidth = mPadPlane->getWidthIPad();
+  int side = hcid % 2;
+
+  int nBits = constants::NBITSTRKLPOS;
+  float granularity = constants::GRANULARITYTRKLPOS;
+  int nMCMcols = constants::NCOLMCM;
```

Comment:
This is just a cosmetic suggestion, but you can simply add `using namespace o2::trd::constants;` above and then use the constants directly without declaring new variables ;)